com/edocti/jintro/Exercise7.java

```diff
@@ -0,0 +1,84 @@
+import java.util.*;
+
+import java.lang.Math;
+
+public class Exercise7 {

```

Comment:
You implemented all these yourself. Not that it's not possible, but there are already NumberFormats and Locales that deal with this problem. This exercise can be implemented in max 15 lines of code (including class and main()) and I encourage you to do it also like that (keep this one and add a new implementation).